### PR TITLE
website: add desktop download callout

### PR DESCRIPTION
**Pull request by mwickett**
🔍 [Preview - homepage](https://boundary-rnujp27du-hashicorp.vercel.app/) (look at hero)
🔍 [Preview - downloads](https://boundary-rnujp27du-hashicorp.vercel.app/downloads)

Adds a Desktop Client callout to `/downloads` and a CTA button to the homepage hero.

This includes the addition of the `MerchDesktopClient` client that is passed to our `ProductDownloader` via the `merchandisingSlot` prop. There are also data fetching changes to grab the appropriate release and build data from `releases.hashicorp.com` for both the Boundary binary _and_ desktop client.

Finally, this also includes a change to `website/data/version.js` to support specifying the version of both the binary and desktop client.

### CODE REVIEW

File: website/components/homepage-hero/HomepageHero.module.css [hunk 1]
```diff
@@ -14,11 +14,26 @@
...
+        margin-left: 5px;
+        vertical-align: middle;
+        content: ' ';
+        background-image: url('/img/icon-right-arrow.svg');
+        background-size: 20px 20px;
+        height: 20px;
+        width: 20px;
+      }
+    }
+
```
Comment (mwickett) [inline website/components/homepage-hero/HomepageHero.module.css:36]: This is a bit of a miserable hack to style the secondary hero CTA button to the design spec. Our `Hero` component doesn't allow for direct variant theming.

File: website/components/merch-desktop-client/index.jsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+          ))}
+        </div>
+      </div>
+    </div>
+  )
+}
+
+function getFileExtension(filename) {
+  return filename.substring(filename.lastIndexOf('.') + 1, filename.length)
+}
```
Comment (mwickett) [inline website/components/merch-desktop-client/index.jsx:32]: Gets the file extension in what I think is a sane way - the desktop client is available as a `.dmg` or `.zip` package.

File: website/data/version.js [hunk 3]
```diff
@@ -1 +1 @@
-export default '0.2.0'
+export default { boundary: '0.2.0', 'boundary-desktop': '1.0.0' }
```
Comment (mwickett) [inline website/data/version.js:1]: Calling out changes to the version file.

File: website/pages/downloads/index.jsx [hunk 4]
```diff
@@ -90,22 +93,41 @@ export default function DownloadsPage({ releases }) {
...
+      },
+    }).then((res) => res.json()),
+  ])
     .then((result) => {
+      const binaryReleases = result.find(
+        (releases) => releases.name === productSlug
+      )
+      const desktopReleases = result.find(
+        (releases) => releases.name === DESKTOP_BINARY_SLUG
+      )
```
Comment (mwickett) [inline website/pages/downloads/index.jsx:126]: There's probably a smarter way to do this, but this works.

File: website/components/homepage-hero/HomepageHero.module.css (hunk 1 above)
Comment (jescalan) [reply website/components/homepage-hero/HomepageHero.module.css:36]: Should we maybe put in a ticket to address this one? I hate to just leave hacks out like this haha

File: website/data/version.js (hunk 3 above)
Comment (jescalan) [reply website/data/version.js:1]: I might export these are separate variables - this is what is done on other docs sites. like:

```js
export const VERSION = '0.2.0'
export const DESKTOP_VERSION = '1.0.0'
```

**Review by jescalan — APPROVED** [review]

File: website/data/version.js (hunk 3 above)
Comment (mwickett) [reply website/data/version.js]: Done, thank you!

**Comment by mwickett** [comment]
@jescalan If you're happy with this, can you merge? I don't have permissions.

**Review by jescalan — APPROVED** [review]

**Pull request merged**